### PR TITLE
steval_stwinbx1 board: Improve sensor support

### PR DESCRIPTION
**Pull request by avisconti**
Add few other sensors support to [steval_stwinbx1](https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/st/steval_stwinbx1) board and a basic sample to test:
    
   - IIS2MDC: 3-axis magnetometer
   - ISM330DHCX: IMU, 3D accelerometer and 3D gyroscope
   - IIS2DLPC: 3-axis accelerometer
   - IIS2ICLX: 2-axis digital inclinometer

Fixed a couple of things in iis2iclx driver while testing the board.

### CODE REVIEW

**Comment by jeppenodgaard** [comment]
Without reading the sample in detail I wonder if SENSOR_SHELL can do the same?

**Comment by avisconti** [comment]
> Without reading the sample in detail I wonder if SENSOR_SHELL can do the same?

It's not exactly the same. I mean, sensor_shell is more an interactive tool to explore things up, and this sample is more board specific and automatic.
Not sure what the general intention is; whether we would like to stop the proliferation of samples or the more the better...

**Comment by jeppenodgaard** [comment]
> It's not exactly the same. I mean, sensor_shell is more an interactive tool to explore things up, and this sample is more board specific and automatic.

I see. There definitely is an advantage of just flashing the board and it will print all sensor values without any interaction.

> Not sure what the general intention is; whether we would like to stop the proliferation of samples or the more the better...

I guess it is somewhere in between.

If a generic sensor sample that printed all initialized sensor's values periodically I think a `samples/sensor` clean up could be done.

File: samples/boards/steval_stwinbx1/sensors/README.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ .. code-block:: console
+
+    STWIN.box dashboard
+
+    STTS22H: Temperature: 24.4 C
+    IIS2DLPC: Accel (m.s-2): x: -5.590, y: -0.536, z: 8.040
+    IIS2MDC: Magn (gauss): x: 0.420, y: -0.116, z: -0.103
+    IIS2MDC: Temperature: 21.0 C
+    ISM330DHCX: Accel (m.s-2): x: 0.000, y: 5.704, z: 7.982
+    ISM330DHCX: GYro (dps): x: 0.026, y: -0.006, z: -0.008
```
Comment (jeppenodgaard) [inline samples/boards/steval_stwinbx1/sensors/README.rst:71]:
```suggestion
    ISM330DHCX: Gyro (dps): x: 0.026, y: -0.006, z: -0.008
```

File: samples/boards/steval_stwinbx1/sensors/src/main.c [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+		printk("%s: device not ready.\n", led1_gpio.port->name);
+		return -1;
+	}
+	gpio_pin_configure_dt(&led1_gpio, GPIO_OUTPUT_INACTIVE);
+
+	/* output led alternate pattern */
+	for (i = 0; i < 8; i++) {
+		gpio_pin_set_dt(&led0_gpio, ((i % 2) == 0) ? 1 : 0);
+		gpio_pin_set_dt(&led1_gpio, ((i % 2) == 1) ? 1 : 0);
+		k_sleep(K_MSEC(100));
```
Comment (jeppenodgaard) [inline samples/boards/steval_stwinbx1/sensors/src/main.c:265]:
```suggestion
		k_msleep(100);
```

File: samples/boards/steval_stwinbx1/sensors/src/main.c [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+	}
+
+	/* output led in sync pattern */
+	for (i = 0; i < 6; i++) {
+		gpio_pin_set_dt(&led0_gpio, ((i % 2) == 0) ? 1 : 0);
+		gpio_pin_set_dt(&led1_gpio, ((i % 2) == 0) ? 1 : 0);
+		k_sleep(K_MSEC(250));
```
Comment (jeppenodgaard) [inline samples/boards/steval_stwinbx1/sensors/src/main.c:272]:
```suggestion
		k_msleep(250);
```

File: samples/boards/steval_stwinbx1/sensors/src/main.c [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+		printf("IIS2MDC: Temperature: %.1f C\n",
+		       sensor_value_to_double(&iis2mdc_temp));
+
+		/* ism330dhcx */
+		printf("ISM330DHCX: Accel (m.s-2): x: %.3f, y: %.3f, z: %.3f\n",
+			sensor_value_to_double(&ism330dhcx_accel[0]),
+			sensor_value_to_double(&ism330dhcx_accel[1]),
+			sensor_value_to_double(&ism330dhcx_accel[2]));
+
+		printf("ISM330DHCX: GYro (dps): x: %.3f, y: %.3f, z: %.3f\n",
```
Comment (jeppenodgaard) [inline samples/boards/steval_stwinbx1/sensors/src/main.c:406]:
```suggestion
		printf("ISM330DHCX: Gyro (dps): x: %.3f, y: %.3f, z: %.3f\n",
```

File: samples/boards/steval_stwinbx1/sensors/src/main.c [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+#ifdef CONFIG_ISM330DHCX_TRIGGER
+		printk("%d:: ism330dhcx acc trig %d\n", cnt, ism330dhcx_acc_trig_cnt);
+		printk("%d:: ism330dhcx gyr trig %d\n", cnt, ism330dhcx_gyr_trig_cnt);
+#endif
+
+#ifdef CONFIG_IIS2ICLX_TRIGGER
+		printk("%d:: iis2iclx trig %d\n", cnt, iis2iclx_trig_cnt);
+#endif
+
+		k_sleep(K_MSEC(2000));
```
Comment (jeppenodgaard) [inline samples/boards/steval_stwinbx1/sensors/src/main.c:437]:
```suggestion
		k_msleep(2000);
```

**Review by jeppenodgaard — DISMISSED** [review]
LGTM. Only nit comments.

File: samples/boards/steval_stwinbx1/sensors/README.rst (hunk 1 above)
Comment (avisconti) [reply samples/boards/steval_stwinbx1/sensors/README.rst]: acked all the points. Thanks!

**Review by jeppenodgaard — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**